sys/kern/kern_cheri_revoke.c

```diff
@@ -95,6 +95,57 @@ cheri_revoke_fini(struct cheri_revoke_syscall_info * __capability crsi,
 	return (res3);
 }
 
+static int
+sysctl_kern_proc_quarantining(SYSCTL_HANDLER_ARGS)
+{
+	struct vmspace *vm;
+	int *name = (int *)arg1;
+	u_int namelen = arg2;
+	struct proc *p;
+	int error;
+	pid_t pid;
+	int is_quarantining;
+
+	if (namelen != 1)
+		return (EINVAL);
+
+	pid = (pid_t)name[0];
+	p = curproc;
+	if (pid == p->p_pid || pid == 0) {
+		if (!SV_CURPROC_FLAG(SV_CHERI)) {
+			is_quarantining = -1;
+			goto out_nocheri;
+		}
+
+		vm = vmspace_acquire_ref(p);
+		/* Don't lock the map since this is informational */
+		is_quarantining = vm->vm_map.vm_cheri_revoke_quarantining;
+		goto out;
+	}
+
+	error = pget(pid, PGET_WANTREAD, &p);
+	if (error != 0)
+		return (error);
+
+	if (!SV_PROC_FLAG(p, SV_CHERI)) {
+		is_quarantining = -1;
+		PRELE(p);
+		goto out_nocheri;
+	}
+	vm = vmspace_acquire_ref(p);
```

Comment:
`vmspace_acquire_ref()` can fail in principle. I'm not sure if that's possible here given that the pget() call above won't give you an exiting process, but in that case you don't need the ref at all, so I'd check for `vm == NULL` just to be safe.